Unable to add Tailwind in Rails 5.1.x project

Using command from readme file for install TailWind CSS fails in rails project 5.1.6

Tried on a sample [project](https://github.com/grsahil20/railstter/tree/feat/boring)

```
Initailizing tailwind configuration
         run  yarn tailwindcss init --full from "."
yarn run v1.22.4
$ /Users/sgrover/projects/side_projects/twitter_clone_rails/node_modules/.bin/tailwindcss init --full
internal/modules/cjs/loader.js:1088
  throw err;
  ^

Error: Cannot find module 'autoprefixer'
```

Thanks @abhaynikam  for taking time to respond. Here is a full log 

```
~/railstter (branch:master) » gb -r                                                                                                                         sgrover@ZALANDO-61315
  origin/HEAD -> origin/master
  origin/feat/boring
  origin/feat/setup-devise
  origin/feat/tweet-delete
  origin/master
------------------------------------------------------------
~/railstter (branch:master) » git checkout -b feat/boring origin/feat/boring                                                                                sgrover@ZALANDO-61315
Branch 'feat/boring' set up to track remote branch 'feat/boring' from 'origin'.
Switched to a new branch 'feat/boring'
------------------------------------------------------------
~/railstter (branch:feat/boring) » rails generate boring:tailwind:install                                                                                   sgrover@ZALANDO-61315
Adding tailwind package
         run  yarn add tailwindcss @tailwindcss/ui from "."
yarn add v1.22.4
info No lockfile found.
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
warning "tailwindcss > postcss-nested@5.0.3" has unmet peer dependency "postcss@^8.1.13".
warning "@tailwindcss/ui > @tailwindcss/typography@0.2.0" has incorrect peer dependency "tailwindcss@^1.5.0".
warning "@tailwindcss/ui > @tailwindcss/custom-forms@0.2.1" has incorrect peer dependency "tailwindcss@^1.0".
warning " > tailwindcss@2.0.2" has unmet peer dependency "autoprefixer@^10.0.2".
warning " > tailwindcss@2.0.2" has unmet peer dependency "postcss@^8.0.9".
warning " > @tailwindcss/ui@0.7.2" has incorrect peer dependency "tailwindcss@^1.8.3".
[4/4] 🔨  Building fresh packages...

success Saved lockfile.
warning Your current version of Yarn is out of date. The latest version is "1.22.5", while you're on "1.22.4".
info To upgrade, run the following command:
$ brew upgrade yarn
success Saved 70 new dependencies.
info Direct dependencies
├─ @tailwindcss/ui@0.7.2
└─ tailwindcss@2.0.2
info All dependencies
├─ @fullhuman/postcss-purgecss@3.1.3
├─ @tailwindcss/custom-forms@0.2.1
├─ @tailwindcss/typography@0.2.0
├─ @tailwindcss/ui@0.7.2
├─ acorn-node@1.8.2
├─ acorn-walk@7.2.0
├─ acorn@7.4.1
├─ ansi-styles@4.3.0
├─ at-least-node@1.0.0
├─ balanced-match@1.0.0
├─ brace-expansion@1.1.11
├─ bytes@3.1.0
├─ camelcase-css@2.0.1
├─ chalk@4.1.0
├─ color-convert@1.9.3
├─ color-name@1.1.4
├─ color-string@1.5.4
├─ color@3.1.3
├─ colorette@1.2.1
├─ commander@6.2.1
├─ concat-map@0.0.1
├─ css-unit-converter@1.1.2
├─ cssesc@3.0.0
├─ defined@1.0.0
├─ detective@5.2.0
├─ didyoumean@1.2.1
├─ escape-string-regexp@1.0.5
├─ fs-extra@9.0.1
├─ fs.realpath@1.0.0
├─ function-bind@1.1.1
├─ glob@7.1.6
├─ graceful-fs@4.2.4
├─ has-flag@3.0.0
├─ has@1.0.3
├─ hex-rgb@4.2.0
├─ html-tags@3.1.0
├─ indexes-of@1.0.1
├─ inflight@1.0.6
├─ inherits@2.0.4
├─ is-arrayish@0.3.2
├─ is-core-module@2.2.0
├─ jsonfile@6.1.0
├─ lodash.toarray@4.4.0
├─ lodash@4.17.20
├─ mini-svg-data-uri@1.2.3
├─ minimatch@3.0.4
├─ minimist@1.2.5
├─ modern-normalize@1.0.0
├─ nanoid@3.1.20
├─ node-emoji@1.10.0
├─ object-assign@4.1.1
├─ object-hash@2.1.1
├─ path-is-absolute@1.0.1
├─ path-parse@1.0.6
├─ postcss-functions@3.0.0
├─ postcss-js@3.0.3
├─ postcss-nested@5.0.3
├─ postcss@8.2.4
├─ pretty-hrtime@1.0.3
├─ purgecss@3.1.3
├─ reduce-css-calc@2.1.8
├─ resolve@1.19.0
├─ simple-swizzle@0.2.2
├─ supports-color@5.5.0
├─ tailwindcss@2.0.2
├─ traverse@0.6.6
├─ uniq@1.0.1
├─ universalify@1.0.0
├─ util-deprecate@1.0.2
└─ xtend@4.0.2
✨  Done in 5.59s.
Initailizing tailwind configuration
         run  yarn tailwindcss init --full from "."
yarn run v1.22.4
$ /Users/sgrover/railstter/node_modules/.bin/tailwindcss init --full
internal/modules/cjs/loader.js:1088
  throw err;
  ^

Error: Cannot find module 'autoprefixer'
Require stack:
- /Users/sgrover/railstter/node_modules/tailwindcss/lib/cli/commands/build.js
- /Users/sgrover/railstter/node_modules/tailwindcss/lib/cli/commands/index.js
- /Users/sgrover/railstter/node_modules/tailwindcss/lib/cli/main.js
- /Users/sgrover/railstter/node_modules/tailwindcss/lib/cli.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:1085:15)
    at Function.Module._load (internal/modules/cjs/loader.js:928:27)
    at Module.require (internal/modules/cjs/loader.js:1145:19)
    at require (internal/modules/cjs/helpers.js:75:18)
    at Object.<anonymous> (/Users/sgrover/railstter/node_modules/tailwindcss/lib/cli/commands/build.js:9:44)
    at Module._compile (internal/modules/cjs/loader.js:1256:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1277:10)
    at Module.load (internal/modules/cjs/loader.js:1105:32)
    at Function.Module._load (internal/modules/cjs/loader.js:967:14)
    at Module.require (internal/modules/cjs/loader.js:1145:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/sgrover/railstter/node_modules/tailwindcss/lib/cli/commands/build.js',
    '/Users/sgrover/railstter/node_modules/tailwindcss/lib/cli/commands/index.js',
    '/Users/sgrover/railstter/node_modules/tailwindcss/lib/cli/main.js',
    '/Users/sgrover/railstter/node_modules/tailwindcss/lib/cli.js'
  ]
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Deprecation warning: Thor exit with status 0 on errors. To keep this behavior, you must define `exit_on_failure?` in `Boring::Tailwind::InstallGenerator`
You can silence deprecations warning by setting the environment variable THOR_SILENCE_DEPRECATION.
The file /Users/sgrover/railstter/postcss.config.js does not appear to exist
Deprecation warning: Thor exit with status 0 on errors. To keep this behavior, you must define `exit_on_failure?` in `Boring::Tailwind::InstallGenerator`
You can silence deprecations warning by setting the environment variable THOR_SILENCE_DEPRECATION.
------------------------------------------------------------
```

I can still see the issue.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,6 +1,7 @@
 # Changelog
 
 ## master (unreleased)
+* Upgrade the Tailwind CSS generator to support Tailwind V2.0. ([@abhaynikam][])
 
 ## 0.6.0 (January 10th, 2021)
 * Updates default ruby version of GitHub Actions install generator. ([@jamesglover][])

--- a/lib/generators/boring/tailwind/install/install_generator.rb
+++ b/lib/generators/boring/tailwind/install/install_generator.rb
@@ -6,18 +6,12 @@ module Boring
       desc "Adds Tailwind CSS to the application"
       source_root File.expand_path("templates", __dir__)
 
-      class_option :skip_tailwind_css_ui,    type: :boolean, aliases: "-sui",
-                                             desc: "Skip adding @tailwindcss/ui package"
       class_option :skip_tailwind_init_full, type: :boolean, aliases: "-sif",
                                              desc: "Skip running tailwindcss init with --full option"
 
       def add_tailwind_package
         say "Adding tailwind package", :green
-        if options[:skip_tailwind_css_ui]
-          run "yarn add tailwindcss"
-        else
-          run "yarn add tailwindcss @tailwindcss/ui"
-        end
+        run "yarn add tailwindcss@latest postcss@latest autoprefixer@latest"
       end
 
       def create_tailwind_config
@@ -54,9 +48,12 @@ module Boring
 
       def insert_stylesheet_in_the_application
         if File.exist?("app/javascript/packs/application.js")
-          append_to_file "app/javascript/packs/application.js" do
-            'import "stylesheets/application"'
-          end
+          stylesheet_tailwind_imports = <<~RUBY
+            \n
+            import "stylesheets/application"
+          RUBY
+
+          append_to_file "app/javascript/packs/application.js", stylesheet_tailwind_imports
         else
           say <<~WARNING, :red
             ERROR: Looks like the webpacker installation is incomplete. Could not find application.js in app/javascript/packs.
@@ -68,6 +65,10 @@ module Boring
         insert_into_file "app/views/layouts/application.html.erb", <<~RUBY, after: /stylesheet_link_tag.*\n/
           \t\t<%= stylesheet_pack_tag 'application', 'data-turbolinks-track': 'reload' %>
         RUBY
+      end
+
+      def show_readme
+        readme "README"
       end
     end
   end

--- a/lib/generators/boring/tailwind/install/templates/README
+++ b/lib/generators/boring/tailwind/install/templates/README
@@ -1,0 +1,13 @@
+===============================================================================
+
+Some setup you must do manually if you haven't yet:
+
+  Current webpacker version relies on the older version of PostCSS.
+  Tailwind 2.0 doesn't support the older version of PostCSS. Please
+  update package.json to update webpacker version.
+
+    ```
+      "@rails/webpacker": "rails/webpacker#b6c2180",
+    ```
+
+===============================================================================

--- a/test/generators/tailwind_install_generator_test.rb
+++ b/test/generators/tailwind_install_generator_test.rb
@@ -19,7 +19,6 @@ class TailwindInstallGeneratorTest < Rails::Generators::TestCase
       quietly { run_generator }
 
       assert_file "package.json" do |content|
-        assert_match(/@tailwindcss\/ui/, content)
         assert_match(/tailwindcss/, content)
       end
 
@@ -49,17 +48,6 @@ class TailwindInstallGeneratorTest < Rails::Generators::TestCase
 
       assert_file "app/views/layouts/application.html.erb" do |content|
         assert_match(/stylesheet_pack_tag \'application\'/, content)
-      end
-    end
-  end
-
-  def test_should_skip_adding_tailwindcss_ui_library_when_used_option_skip_tailwind_css_ui
-    Dir.chdir(app_path) do
-      quietly { run_generator [destination_root, "--skip_tailwind_css_ui"] }
-
-      assert_file "package.json" do |content|
-        assert_no_match(/@tailwindcss\/ui/, content)
-        assert_match(/tailwindcss/, content)
       end
     end
   end

